lib/Selenium/Chrome.pm

```diff
@@ -10,9 +15,120 @@ extends 'Selenium::Remote::Driver';
 
 =cut
 
+my $default_binary_server = '127.0.0.1';
+my $default_binary_port = 9515;
+
 has '+browser_name' => (
     is => 'ro',
     default => sub { 'chrome' }
 );
 
+
+# By shadowing the parent's port function, we can set the port in
+# _build_binary_mode's builder
+has '+port' => (
+    is => 'lazy'
+);
+
+has 'binary_mode' => (
+    is => 'ro',
+    init_arg => undef,
+    builder => 1
+);
+
+sub _build_binary_mode {
+    my ($self) = @_;
+
+    if (! $self->has_remote_server_addr && ! $self->has_port) {
+        my $executable = _find_executable('chromedriver');
+        my $port = _find_open_port_above($default_binary_port);
+        my $command = _construct_command($executable, $port);
+
+        system($command);
+        my $success = wait_until { _probe_port($port) } timeout => 10;
+        if ($success) {
+            $self->port($port);
+        }
+        else {
+            warn qq(
+Unable to start up the chromedriver binary via:
+
+    $command
+
+We'll try falling back to standard Remote Driver mode via the webdriver.chrome.driver property...
+);
+
+        }
+
+        return 1;
+    }
+    else {
+        return 0;
+    }
+}
+
+sub _find_executable {
+    my ($binary) = @_;
+
+    my $executable = which($binary);
+
+    if (not defined $executable) {
+        warn q(
```

Comment:
Not entirely sure what to do here, or in the other warn case above. The other bindings get to `die` when they can't find the binary, because their default behavior is to depend on the binary, or be told what port to find it on.

Unfortunately our default behavior is to assume localhost:4444/wd/hub, so currently I'm _not_ dying so that we can fall back to S:R:D's defaults. It's entirely possible that this will be confusing and frustrating behavior, though.